### PR TITLE
Add new pop-up modal when image is clicked

### PR DESCRIPTION
**Pull request by PrabSG**
Create a separate modal for displaying the image in a larger size, as well as an accompanying caption, for use on portfolio page's profile image currently. Redesign on header and content to use the more dynamic flexbox layout system.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -26,3 +26,21 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Opens a modal with the given image and caption.
+ */
+function openSimpleImageModal(modalId, modalImgId, captionId, closeId, imgSrc, captionText) {  
+  var modal = document.getElementById(modalId);
```
Comment (DarthGandalf) [inline portfolio/src/main/webapp/script.js:34]: let modal = ...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

**Review by DarthGandalf — CHANGES_REQUESTED** [review]
The first 3 commits here are better as a single commit, because they add one feature, and after first commit the webpage doesn't even work properly

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (DarthGandalf) [reply portfolio/src/main/webapp/script.js:34]: https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -26,3 +26,21 @@ function addRandomGreeting() {
...
+  var modalImg = document.getElementById(modalImgId);
+  var caption = document.getElementById(captionId);
+
+  modal.style.display = "block";
+  modalImg.src = imgSrc;
+  caption.innerHTML = captionText;
```
Comment (DarthGandalf) [inline portfolio/src/main/webapp/script.js:40]: Use `innerText` instead

innerHTML is a very easy way to get XSS

Comment (DarthGandalf) [reply portfolio/src/main/webapp/script.js:40]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (PrabSG) [reply portfolio/src/main/webapp/script.js:34]: Ah yes will change this, is it okay to use const given it is part of ES6?

Comment (DarthGandalf) [reply portfolio/src/main/webapp/script.js:34]: sure

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (PrabSG) [reply portfolio/src/main/webapp/script.js:40]: Will keep this in mind for other sections, thanks. Done

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (PrabSG) [reply portfolio/src/main/webapp/script.js:34]: Done

**Comment by DarthGandalf** [comment]
please use `git rebase -i master` to fix number of commits (and distribution of code between commits), then `git push -f`

**Comment by PrabSG** [comment]
> please use `git rebase -i master` to fix number of commits (and distribution of code between commits), then `git push -f`

Squashed commits into 1 and renamed appropriately, done.

**Review by DarthGandalf — APPROVED** [review]

**Pull request merged**